server/src/com/gip/xyna/xprc/xfractwfe/python/PythonMdmGeneration.java

```diff
@@ -79,58 +74,70 @@ public String createPythonMdm(Long revision, boolean withImpl, boolean typeHints
     GenerationBaseCache cache = new GenerationBaseCache();
-    List<XynaObjectInformation> result = new ArrayList<XynaObjectInformation>(objects.size());
+    Map<String, XynaObjectInformation> map = new HashMap<>();
+    map.put("XynaObject", null);
+    map.put("XynaException", null);
     for (XMOMDatabaseSearchResultEntry obj : objects) {
-      if(obj.getType().equals(XMOMDatabaseType.SERVICEGROUP)) {
+      if (obj.getType().equals(XMOMDatabaseType.SERVICEGROUP)) {
         String dtFqn = obj.getFqName().substring(0, obj.getFqName().lastIndexOf('.'));
-        if(objects.stream().anyMatch(x -> x.getFqName().equals(dtFqn))) {
+        if (objects.stream().anyMatch(x -> x.getFqName().equals(dtFqn))) {
           continue; // we discover all methods by examining the datatype
         } else {
           obj.setFqName(dtFqn);
         }
       }
       XynaObjectInformation info = loadXynaObjectInfo(revision, obj.getFqName(), obj.getType().equals(XMOMDatabaseType.EXCEPTION), cache);
-      result.add(info);
+      map.put(info.fqn, info);
     }
+    return map;
+  }
+  
+  private List<XynaObjectInformation> convertAndSortSearchResult(List<XMOMDatabaseSearchResultEntry> objects, Long revision) {
+    Map<String, XynaObjectInformation> map = loadObjects(objects, revision);
+    List<XynaObjectInformation> result = new ArrayList<XynaObjectInformation>(objects.size());
 
-    Set<String> availableClasses = new HashSet<String>();
-    availableClasses.add("XynaObject");
-    availableClasses.add("XynaException");
-    //sort - start with Objects without parent
-    int reorders = 0;
-    int breaker = 0;
-    for (int i = 0; i < result.size(); i++) {
-      XynaObjectInformation obj = result.get(i);
-      if(!availableClasses.contains(obj.parent)) {
-        result.remove(i);
-        result.add(obj);
-        i--;
-        breaker++;
-        reorders++;
-        if(breaker >= result.size()) {
-          throw new RuntimeException("Could not order objects. " + i + " of " + objects.size() + " ordered. Current object: " + obj.fqn);
-        }
-      } else {
-        availableClasses.add(convertToPythonFqn(obj.fqn));
-        breaker = 0;
+    for (String fqn: map.keySet()) {
+      XynaObjectInformation obj = map.get(fqn);
+      if (obj == null) {
+        continue;
       }
+      addObjectToResult(obj, map, result);
     }
-    
-    if(logger.isDebugEnabled()) {
-      logger.debug("sorted through " + objects.size() + " objects and reordered " + reorders + " times.");
-    }
-    
+
     return result;
   }
+  
+
+  private void addObjectToResult(XynaObjectInformation obj, Map<String, XynaObjectInformation> map, List<XynaObjectInformation> result) {
+    Stack<XynaObjectInformation> hierarchy = new Stack<>();
+    hierarchy.push(obj);
+
+    while (map.containsKey(hierarchy.peek().parent) && map.get(hierarchy.peek().parent) != null) {
```

Comment:
map.get(hierarchy.peek().parent) returns nulle if map.containsKey(hierarchy.peek().parent) is false.
Therfore it is equalivent to 
 while (map.get(hierarchy.peek().parent) != null) {